### PR TITLE
[WIP] Add --node-labels flag to cluster create

### PR DESCRIPTION
**Pull request by aknuds1**
**NOTE: THIS IS WIP!**

I need to [add labels to Kubernetes nodes](https://github.com/kubernetes/kops/blob/master/docs/labels.md) as they are created with the cluster, but there's no current command line support for it unlike for cloud labels. This PR adds a CLI flag `--node-labels` to `kops create cluster` in order to be able to specify node labels.

The patch compiles at the moment, but I'm not sure how to actually implement the application of the labels to the nodes as they are created.

### CODE REVIEW

File: pkg/apis/kops/v1alpha1/cluster.go [hunk 1]
```diff
@@ -236,6 +236,7 @@ type ClusterSpec struct {
 	Authorization *AuthorizationSpec `json:"authorization,omitempty"`
 
 	// Tags for AWS instance groups
+	NodeLabels map[string]string `json:"nodeLabels,omitempty"`
```
Comment (justinsb) [inline pkg/apis/kops/v1alpha1/cluster.go]: FYI comment went out-of-order here

**Comment by justinsb** [comment]
No intrinsic objection to this, but this would mean the same node-labels on all the nodes, right?  Can you explain what the use case is?  I think we don't have this because if all the nodes have the same labels, the label can't be used to differentiate them...

**Comment by aknuds1** [comment]
@justinsb I guess nodes in a kops context means only worker machines (although this is confusing as `kubectl get nodes` returns also masters), but my use case is anyhow that I need to label all of those `beta.kubernetes.io/fluentd-ds-ready=true` in order for the Fluentd DaemonSet to execute on them ([this is standard behaviour as of 1.6](https://github.com/kubernetes/features/blob/master/release-1.6/release-notes-draft.md#fluentd)).

File: pkg/apis/kops/v1alpha1/cluster.go (hunk 1 above)
Comment (chrislovecnm) [inline pkg/apis/kops/v1alpha1/cluster.go]: Why are we adding an object since we already have an object like this in instance groups?  I am confused.

**Comment by aknuds1** [comment]
@chrislovecnm Actually I'm not sure how to go about this; I'm new to Go programming and if you look at my patch description, you'll see that I could use advice in how to apply the labels to nodes as they get created.

**Pull request closed without merging**